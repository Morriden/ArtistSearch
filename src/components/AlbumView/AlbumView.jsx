import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const AlbumView = ({ releases, bandName }) => {
  const artistReleases = releases.map(release => (
    <li key={release.id}>
      <Link to={`/album/${bandName}/${release.id}`}>
        <div>
          {release.title}, {release.date}
          <img src={`http://coverartarchive.org/release/${release.id}/front`} alt='Image Not Found'/>
        </div>
      </Link>
    </li>
  ));

  return (
    <section>
      {artistReleases}
    </section>   
  );
};

AlbumView.propTypes = {
  releases: PropTypes.arrayOf(PropTypes.object).isRequired,
  bandName: PropTypes.string.isRequired
};

export default AlbumView;
