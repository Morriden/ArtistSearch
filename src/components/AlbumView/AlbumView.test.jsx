import React from 'react';
import { shallow } from 'enzyme';
import AlbumView from './AlbumView';

describe('Displays Releases for and Artist', () => {
  it('Should be the releases from an Artist', () => {
    const wrapper = shallow(<AlbumView bandName={'im a name'} releases={[{
      title: 'title1', date: 'somedate', id: 'ID1'
    },
    {
      title: 'title2', date: 'somedate', id: 'ID2'
    }, {
      title: 'title3', date: 'somedate', id: 'ID3'
    }]}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
