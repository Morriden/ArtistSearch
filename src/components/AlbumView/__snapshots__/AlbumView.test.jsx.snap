// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays Releases for and Artist Should be the releases from an Artist 1`] = `
<section>
  <li
    key="ID1"
  >
    <Link
      to="/album/im a name/ID1"
    >
      <div>
        title1
        , 
        somedate
        <img
          alt="Image Not Found"
          src="http://coverartarchive.org/release/ID1/front"
        />
      </div>
    </Link>
  </li>
  <li
    key="ID2"
  >
    <Link
      to="/album/im a name/ID2"
    >
      <div>
        title2
        , 
        somedate
        <img
          alt="Image Not Found"
          src="http://coverartarchive.org/release/ID2/front"
        />
      </div>
    </Link>
  </li>
  <li
    key="ID3"
  >
    <Link
      to="/album/im a name/ID3"
    >
      <div>
        title3
        , 
        somedate
        <img
          alt="Image Not Found"
          src="http://coverartarchive.org/release/ID3/front"
        />
      </div>
    </Link>
  </li>
</section>
`;
